agent/app/agent.go

```diff
@@ -108,6 +110,26 @@ func newAgent(
 		return nil, err
 	}
 
+	var metadataManager containermetadata.MetadataManager
+	if cfg.ContainerMetadataEnabled {
+		// Get the oldest Docker Client version with up to date inspect API
+		// as some metadata is unavailable in older API client versions
+		// If this version is unavailable we use default
+		// Due to limitations of interfaces and import cycles
+		// we can not do this logic in containermetadata package currently
+		// If we move DockerClient out of the engine, this section should be moved
+		// into containermetadata.NewMetadataManager()
+		supportedClients := dockerClient.SupportedVersions()
+		metadataDockerClient := dockerClient
+		for index := range supportedClients {
+			if supportedClients[index] == containermetadata.ContainerMetadataClientVersion {
+				metadataDockerClient = dockerClient.WithVersion(containermetadata.ContainerMetadataClientVersion)
+				break
+			}
+		}
+		metadataManager = containermetadata.NewMetadataManager(metadataDockerClient, cfg)
```

Comment:
The client version matters because with Task ENI multiple network modes are now supported. The default client does not give information on additional network modes which is only available in v1.21.